### PR TITLE
fix: ENCODED_SIGNER_KEY in encryption context

### PR DESCRIPTION
**Pull request by seebees**
A CryptographicMaterialsManager can change entries to the encryptionContext
but changing these values has consequences.
The DefaultCryptographicMaterialsManager uses the value in the encryption context
to store public signing key.
If the caller is using this value in their encryption context the Default CMM
is probably not the CMM they want to use.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: modules/material-management-browser/src/browser_cryptographic_materials_manager.ts [hunk 1]
```diff
@@ -49,11 +49,23 @@ export class WebCryptoDefaultCryptographicMaterialsManager implements WebCryptoM
   async getEncryptionMaterials ({ suite, encryptionContext }: WebCryptoEncryptionRequest): Promise<WebCryptoEncryptionMaterial> {
     suite = suite || new WebCryptoAlgorithmSuite(AlgorithmSuiteIdentifier.ALG_AES256_GCM_IV12_TAG16_HKDF_SHA384_ECDSA_P384)
 
+    /* Precondition: WebCryptoDefaultCryptographicMaterialsManager should reserve the ENCODED_SIGNER_KEY constant from @aws-crypto/serialize.
```
Comment (mattsb42-aws) [inline modules/material-management-browser/src/browser_cryptographic_materials_manager.ts]:
```suggestion
    /* Precondition: WebCryptoDefaultCryptographicMaterialsManager must reserve the ENCODED_SIGNER_KEY constant from @aws-crypto/serialize.
```

File: modules/material-management-node/src/node_cryptographic_materials_manager.ts [hunk 2]
```diff
@@ -49,11 +49,23 @@ export class NodeDefaultCryptographicMaterialsManager implements NodeMaterialsMa
   async getEncryptionMaterials ({ suite, encryptionContext }: NodeEncryptionRequest): Promise<NodeEncryptionMaterial> {
     suite = suite || new NodeAlgorithmSuite(AlgorithmSuiteIdentifier.ALG_AES256_GCM_IV12_TAG16_HKDF_SHA384_ECDSA_P384)
 
+    /* Precondition: NodeDefaultCryptographicMaterialsManager should reserve the ENCODED_SIGNER_KEY constant from @aws-crypto/serialize.
```
Comment (mattsb42-aws) [inline modules/material-management-node/src/node_cryptographic_materials_manager.ts]:
```suggestion
    /* Precondition: NodeDefaultCryptographicMaterialsManager must reserve the ENCODED_SIGNER_KEY constant from @aws-crypto/serialize.
```

File: modules/material-management-browser/src/browser_cryptographic_materials_manager.ts (hunk 1 above)
Comment (seebees) [reply modules/material-management-browser/src/browser_cryptographic_materials_manager.ts]: fixed, and the test is updated...

File: modules/material-management-node/src/node_cryptographic_materials_manager.ts (hunk 2 above)
Comment (seebees) [reply modules/material-management-node/src/node_cryptographic_materials_manager.ts]: fixed, and the test is updated...

**Pull request merged**